cli/removemfadevice.go

```diff
@@ -0,0 +1,83 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/99designs/keyring"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/99designs/aws-vault/mfa"
+	"github.com/99designs/aws-vault/vault"
+)
+
+type RemoveMfaDeviceCommandInput struct {
+	ProfileName string
+	Keyring     keyring.Keyring
+	Config      vault.Config
+	Username    string
+}
+
+func ConfigureRemoveMfaCommand(app *kingpin.Application) {
+	input := RemoveMfaDeviceCommandInput{}
+
+	cmd := app.Command("remove-mfa-device", "Removes the MFA device associated with the IAM user")
+	cmd.Arg("profile", "Name of the profile").
+		Required().
+		HintAction(awsConfigFile.ProfileNames).
+		StringVar(&input.ProfileName)
+
+	cmd.Arg("username", "Name of the user to add the MFA as device for").
+		Required().
+		StringVar(&input.Username)
+
+	cmd.Action(func(c *kingpin.ParseContext) error {
+		input.Keyring = keyringImpl
+		input.Config.MfaTokenProvider = GlobalFlags.MfaTokenProvider
+		RemoveMFADeviceCommand(app, input)
+		return nil
+	})
+
+}
+
+func RemoveMFADeviceCommand(app *kingpin.Application, input RemoveMfaDeviceCommandInput) {
+	p, found := awsConfigFile.ProfileSection(input.ProfileName)
+	if !found {
+		app.Fatalf("Profile with name '%s' not found")
+	}
+
+	provider := &vault.KeyringProvider{
+		Keyring:         &vault.CredentialKeyring{Keyring: input.Keyring},
+		CredentialsName: p.Name,
+	}
+
+	creds, err := provider.Retrieve()
+	if err != nil {
+		app.Fatalf("unable to retrieve creds for profile %s", p.Name)
+	}
+
+	sess, err := session.NewSessionWithOptions(session.Options{
+		Config: aws.Config{
+			Region:      aws.String(p.Region),
+			Credentials: credentials.NewCredentials(&credentials.StaticProvider{Value: creds}),
+		},
+	})
+	if err != nil {
+		app.Fatalf("error creating session: %s", err.Error())
+	}
```

Comment:
You can use vault.NewSession for this
https://github.com/99designs/aws-vault/blob/f26b718f741a3f5c51809982539e6d7a1b6f4280/vault/vault.go#L20

Reply:
done (I think it's as you expect but not totally sure, please check)